pkg/storegateway/indexheader/indexheaderpb/samples.proto

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+syntax = "proto3";
+
+package indexheaderpb;
+
+option go_package = "indexheaderpb";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+// Do not generate XXX fields to reduce memory footprint.
+option (gogoproto.goproto_unkeyed_all) = false;
+option (gogoproto.goproto_unrecognized_all) = false;
+option (gogoproto.goproto_sizecache_all) = false;
+
+message Samples {
+	Symbols symbols = 1;
+	PostingOffsetTable postingsOffsetTable = 2;
+}
+
+message Symbols {
+	repeated int64 offsets = 1 [packed=true];
+	int64 symbolsCount = 2;
+}
+
+message PostingOffsetTable {
+	// Postings is a map of label names -> postingValueOffsets
+	map<string, postingValueOffsets> postings = 1;
+}
+
+// postingValueOffsets stores a list of the first, last, and every 32nd (config default) postingOffset for this label name.
+message postingValueOffsets {
```

Comment:
These appear to be exported anyway in the generated code so no need for them to start with a lowercase letter.